cmd/proxy/mocks_test.go

```diff
@@ -0,0 +1,253 @@
+}
+
+// CancelRead aborts receiving on this stream.
+// It will ask the peer to stop transmitting stream data.
+// Read will unblock immediately, and future Read calls will fail.
+// When called multiple times or after reading the io.EOF it is a no-op.
+func (mqs *MockQUICStream) CancelRead(quic.ErrorCode) {}
+
+func (mqs *MockQUICStream) SetReadDeadline(t time.Time) error {
+	return ErrUnimplementedMock
+}
+
+// *********************************
+// quic.SendStream interface methods
+// *********************************
+
+// Write writes data to the stream.
+// Write can be made to time out and return a net.Error with Timeout() == true
+// after a fixed time limit; see SetDeadline and SetWriteDeadline.
+// If the stream was canceled by the peer, the error implements the StreamError
+// interface, and Canceled() == true.
+// If the session was closed due to a timeout, the error satisfies
+// the net.Error interface, and Timeout() will be true.
+func (mqs *MockQUICStream) Write(p []byte) (int, error) {
+	mqs.OutBuf = make([]byte, len(p))
+
+	return copy(mqs.OutBuf, p), nil
+}
+
+// Close closes the write-direction of the stream.
+// Future calls to Write are not permitted after calling Close.
+// It must not be called concurrently with Write.
+// It must not be called after calling CancelWrite.
+func (mqs *MockQUICStream) Close() error {
+	return nil
+}
+
+// The context is canceled as soon as the write-side of the stream is closed.
+// This happens when Close() or CancelWrite() is called, or when the peer
+// cancels the read-side of their stream.
+// Warning: This API should not be considered stable and might change soon.
+func (mqs *MockQUICStream) Context() context.Context {
+	panic(ErrUnimplementedMock)
+}
+
+// CancelWrite aborts sending on this stream.
+// Data already written, but not yet delivered to the peer is not guaranteed to be delivered reliably.
+// Write will unblock immediately, and future calls to Write will fail.
+// When called multiple times or after closing the stream it is a no-op.
+func (mqs *MockQUICStream) CancelWrite(quic.ErrorCode) {}
+
+// SetWriteDeadline sets the deadline for future Write calls
+// and any currently-blocked Write call.
+// Even if write times out, it may return n > 0, indicating that
+// some of the data was successfully written.
+// A zero value for t means Write will not time out.
+func (mqs *MockQUICStream) SetWriteDeadline(t time.Time) error {
+	return ErrUnimplementedMock
+}
+
+// *****************************
+// quic.Stream interface methods
+// *****************************
+
+// SetDeadline sets the read and write deadlines associated
+// with the connection. It is equivalent to calling both
+// SetReadDeadline and SetWriteDeadline.
+func (mqs *MockQUICStream) SetDeadline(t time.Time) error {
+	return ErrUnimplementedMock
+}
+
+// MockQUICSession satisfies the quic.Session interface
+type MockQUICSession struct {
+	Streams []*MockQUICStream
+
+	accepted int
```

Comment:
I do not see this used anywhere. I understand the intention to extend this in the future but I don't wanna merge anything that is not used because it's an extra maintenance burden.